waiter/src/waiter/service_description.clj

```diff
@@ -837,8 +837,14 @@
   "Loads the service description for the specified service-id including any overrides."
   [kv-store service-id service-description-defaults metric-group-mappings &
    {:keys [effective? nil-on-missing? refresh] :or {effective? true nil-on-missing? true refresh false}}]
-  (cond-> (fetch-core kv-store service-id :nil-on-missing? nil-on-missing? :refresh refresh)
-          effective? (default-and-override metric-group-mappings kv-store service-description-defaults service-id)))
+  (let [service-description (fetch-core kv-store service-id :nil-on-missing? nil-on-missing? :refresh refresh)
+        service-description (if (and (empty? service-description) (not refresh))
+                              (do
+                                (log/info "force refreshing fetch of service description for" service-id)
+                                (fetch-core kv-store service-id :nil-on-missing? nil-on-missing? :refresh true))
```

Comment:
Why do we  "force refreshing fetch of service description" when `refresh` is *not* true?

Maybe the logic is backwards. But if that's the case, then why aren't the unit tests failing?

Assuming I'm reading this correctly and the logic *is* inverted, we should add a unit test for this (to verify that it fails as-is), and then fix it.

Reply:
Doesn't "force refresh" indicate we are refreshing even though `refresh` is not true. Maybe we should check `(nil? refresh)` and default to `nil` instead of false for `refresh`? That way a caller can indicate when he doesn't want the force refresh behavior. What do you think?

Reply:
The parameters `nil-on-missing?` and  `refresh` were unspecified from any of the callers, so ended up removing them.